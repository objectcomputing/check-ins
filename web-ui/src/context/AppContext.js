import React, { useEffect, useReducer, useMemo } from "react";
import { createSelector } from 'reselect';
import { getCurrentUser, updateMember, getAllMembers } from "../api/member";
import { getCheckinByMemberId, createCheckin } from "../api/checkins";
import { BASE_API_URL } from "../api/api";
import axios from "axios";

export const MY_PROFILE_UPDATE = "@@check-ins/update_profile";
export const UPDATE_USER_BIO = "@@check-ins/update_bio";
export const UPDATE_CHECKINS = "@@check-ins/update_checkins";
export const UPDATE_INDEX = "@@check-ins/update_index";
export const UPDATE_TOAST = "@@check-ins/update_toast";
export const UPDATE_CURRENT_CHECKIN = "@@check-ins/update_current_checkin";
export const UPDATE_TEAMS = "@@check-ins/update_teams";
export const UPDATE_MEMBER_PROFILES = "@@check-ins/update_member_profiles";
export const UPDATE_TEAM_MEMBERS = "@@check-ins/update_team_members";
export const UPDATE_SELECTED_PROFILE = "@@check-ins/update_selected_profile";
const SET_CSRF = "@@check-ins/update_csrf";

const AppContext = React.createContext();

const reducer = (state, action) => {
  switch (action.type) {
    case MY_PROFILE_UPDATE:
      state.userProfile = action.payload;
      break;
    case UPDATE_USER_BIO:
      state.userProfile.memberProfile.bioText = action.payload;
      updateMember(state.userProfile.memberProfile);
      break;
    case UPDATE_CHECKINS:
      state.checkins = action.payload;
      //sort by date
      state.checkins.sort(function (a, b) {
        return new Date(...a.checkInDate) - new Date(...b.checkInDate);
      });
      state.currentCheckin = state.checkins[state.checkins.length - 1];
      break;
    case SET_CSRF:
      state.csrf = action.payload;
      break;
    case UPDATE_TOAST:
      state.toast = action.payload;
      break;
    case UPDATE_TEAMS:
      state.teams = action.payload;
      break;
    case UPDATE_MEMBER_PROFILES:
      state.memberProfiles = action.payload;
      break;
    case UPDATE_TEAM_MEMBERS:
      state.teamMembers = action.payload;
      break;
    case UPDATE_CURRENT_CHECKIN:
      state.currentCheckin = action.payload;
      break;
    case UPDATE_SELECTED_PROFILE:
      const { payload } = action;
      if (state.selectedProfile !== payload) {
        state.selectedProfile = payload;
        state.currentCheckin = payload ? payload.checkIn : {};
      }
      if (payload === undefined) {
        state.checkins = [];
      }
      break;
    default:
  }
  return { ...state };
};

const initialState = {
  checkins: [],
  csrf: undefined,
  currentCheckin: {},
  teams: [],
  memberProfiles: [],
  index: 0,
  selectedProfile: undefined,
  toast: {
    severity: "",
    toast: "",
  },
  userProfile: undefined,
};

const getCheckins = async (id, pdlId, date, dispatch, csrf) => {
  const res = await getCheckinByMemberId(id, csrf);
  let data =
    res.payload && res.payload.data && res.payload.status === 200 && !res.error
      ? res.payload.data
      : null;
  if (data && data.length > 0) {
    const allComplete = data.every((checkin) => checkin.completed === true);
    if (allComplete) {
      const prevCheckinDate = data[data.length - 1].checkInDate;
      if (pdlId) {
        const res = await createCheckin({
          teamMemberId: id,
          pdlId: pdlId,
          checkInDate: date(3, prevCheckinDate),
          completed: false,
        });
        const checkin =
          res.payload && res.payload.data && !res.error
            ? res.payload.data
            : null;
        data.push(checkin);
      }
    }
  } else if (data && data.length === 0) {
    if (pdlId) {
      const res = await createCheckin({
        teamMemberId: id,
        pdlId: pdlId,
        checkInDate: date(1),
        completed: false,
      });
      const checkin =
        res.payload && res.payload.data && !res.error ? res.payload.data : null;
      data = [checkin];
    }
  }
  //without this check you get infinite checkin calls
  if (data.length > 0) {
    dispatch({ type: UPDATE_CHECKINS, payload: data });
  }
};

const AppContextProvider = (props) => {
  const [state, dispatch] = useReducer(reducer, initialState);
  const memberProfile =
    state && state.userProfile && state.userProfile.memberProfile
      ? state.userProfile.memberProfile
      : undefined;
  const id = memberProfile ? memberProfile.id : undefined;
  const selectedProfile = state && state.selectedProfile;
  const selectedId = selectedProfile ? selectedProfile.id : undefined;


  const { csrf } = state;

  useEffect(() => {
    const getCsrf = async () => {
      if (!csrf) {
        const res = await axios({
          url: `${BASE_API_URL}/csrf/cookie`,
          responseType: "text",
          withCredentials: true,
        });
        if (res && res.data) {
          dispatch({ type: SET_CSRF, payload: res.data });
        }
      }
    };
    getCsrf();
  }, [csrf]);

  useEffect(() => {
    const updateUserProfile = async () => {
      let res = await getCurrentUser(csrf);
      let profile =
        res.payload && res.payload.data && !res.error
          ? res.payload.data
          : undefined;

      if (profile) {
        dispatch({ type: MY_PROFILE_UPDATE, payload: profile });
      }
    };
    if (csrf) {
      updateUserProfile();
    }
  }, [csrf]);

  const date = (months, prevCheckinDate) => {
    let currentMonth = new Date().getMonth() + 1;
    let newDate = prevCheckinDate ? new Date(...prevCheckinDate) : new Date();
    if (prevCheckinDate) {
      let prevCheckinMonth = prevCheckinDate.getMonth() + 1;
      newDate.setMonth(
        newDate.getMonth() + currentMonth - prevCheckinMonth >= 3 ? 1 : months
      );
    } else {
      newDate.setMonth(newDate.getMonth() + months);
    }
    const year = newDate.getFullYear();
    const month = newDate.getMonth() + 1;
    const day = newDate.getDate();
    const hours = newDate.getHours();
    const minutes = newDate.getMinutes();
    const dateTimeArray = [year, month, day, hours, minutes, 0];
    return dateTimeArray;
  };

  useEffect(() => {
    async function getMemberProfiles() {
      let res = await getAllMembers(csrf);
      let profiles =
        res.payload && res.payload.data && !res.error
          ? res.payload.data
          : undefined;

      if (profiles) {
        dispatch({ type: UPDATE_MEMBER_PROFILES, payload: profiles });
      }
    }
    if (csrf) {
      getMemberProfiles();
    }
  }, [csrf]);

  useEffect(() => {
    if (selectedId && csrf) {
      getCheckins(selectedId, id, date, dispatch, csrf);
    }
  }, [csrf, selectedId, id]);

  const value = useMemo(() => {
    return { state, dispatch };
  }, [state]);
  return (
    <AppContext.Provider value={props.value || value}>
      {props.children}
    </AppContext.Provider>
  );
};

export const selectMemberProfiles = state => state.memberProfiles;
export const selectTeamMembers = state => state.teamMembers;

export const selectProfileMap = createSelector(
  selectMemberProfiles,
  (memberProfiles) => {
    if (memberProfiles && memberProfiles.length) {
      memberProfiles = memberProfiles.reduce((mappedById, profile) => {
        mappedById[profile.id] = profile;
        return mappedById;
      }, {});
    }
    return memberProfiles;
  }
);

export const selectMembersByTeamId = id => createSelector(
  selectTeamMembers,
  (teamMembers) => {
    let members = [];
    if (teamMembers && teamMembers.length) {
      members = teamMembers.filter((member) => member.teamid === id);
    }
    return members;
  }
);

export const selectMemberProfilesByTeamId = id => createSelector(
  selectMembersByTeamId(id),
  selectProfileMap,
  (members, profileMap) => members.map((member) => {
      return { ...profileMap[member.memberid], ...member };
    })
);

export { AppContext, AppContextProvider };
