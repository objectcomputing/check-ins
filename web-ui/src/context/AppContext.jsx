import React, { useEffect, useReducer, useMemo } from 'react';
import { reducer, initialState } from './reducer';
import { getCheckins, getAllCheckinsForAdmin } from './thunks';
import {
  MY_PROFILE_UPDATE,
  SET_CSRF,
  SET_ROLES,
  SET_USER_ROLES,
  UPDATE_GUILDS,
  UPDATE_MEMBER_SKILLS,
  UPDATE_MEMBER_PROFILES,
  UPDATE_TERMINATED_MEMBERS,
  UPDATE_SKILLS,
  UPDATE_CERTIFICATIONS,
  UPDATE_TEAMS,
  UPDATE_PEOPLE_LOADING,
  UPDATE_TEAMS_LOADING
} from './actions';
import {
  getCurrentUser,
  getAllMembers,
  getAllTerminatedMembers
} from '../api/member';
import {
  selectCanViewCheckinsPermission,
  selectCanViewTerminatedMembers,
} from './selectors';
import { getAllRoles, getAllUserRoles } from '../api/roles';
import { getMemberSkills } from '../api/memberskill';
import { BASE_API_URL } from '../api/api';
import { getAllGuilds } from '../api/guild';
import { getSkills } from '../api/skill';
import { getAllTeams } from '../api/team';
import {getCertifications} from "../api/certification.js";

const AppContext = React.createContext();

function getSessionCookieValue(name) {
  const cookies = document?.cookie?.split(';');
  for (let i = 0; i < cookies.length; i++) {
    const cookie = cookies[i].trim();
    if (cookie.startsWith(name + '=')) {
      return decodeURIComponent(cookie.substring(name.length + 1));
    }
  }
  return null;
}

const AppContextProvider = props => {
  const [state, dispatch] = useReducer(
    reducer,
    props?.value?.state || initialState
  );
  const userProfile =
    state && state.userProfile ? state.userProfile : undefined;
  const memberProfile =
    userProfile && userProfile.memberProfile
      ? userProfile.memberProfile
      : undefined;

  const id = memberProfile ? memberProfile.id : undefined;
  const pdlId = memberProfile ? memberProfile.pdlId : undefined;
  const {
    csrf,
    guilds,
    teams,
    memberSkills,
    memberProfiles,
    checkins,
    skills,
    certifications,
    roles,
    userRoles
  } = state;
  const url = `${BASE_API_URL}/csrf/cookie`;
  useEffect(() => {
    const getCsrf = async () => {
      if (!csrf) {
        const payload = getSessionCookieValue('_csrf');
        if (payload) {
          dispatch({ type: SET_CSRF, payload });
        } else {
          const res = await fetch(url, {
            responseType: 'text',
            credentials: 'include'
          });
          if (res && res.ok) {
            dispatch({ type: SET_CSRF, payload: await res.text() });
          }
        }
      }
    };
    getCsrf();
  }, [csrf]);

  useEffect(() => {
    async function getGuilds() {
      let res = await getAllGuilds(csrf);
      let data =
        res.payload &&
        res.payload.data &&
        res.payload.status === 200 &&
        !res.error
          ? res.payload.data
          : null;
      if (data) {
        dispatch({ type: UPDATE_GUILDS, payload: data });
      }
    }
    if (csrf && !guilds) {
      getGuilds();
    }
  }, [csrf, guilds]);

  useEffect(() => {
    async function getTeams() {
      let res = await getAllTeams(csrf);
      let data =
        res.payload &&
        res.payload.data &&
        res.payload.status === 200 &&
        !res.error
          ? res.payload.data
          : null;
      if (data) {
        dispatch({ type: UPDATE_TEAMS, payload: data });
        dispatch({ type: UPDATE_TEAMS_LOADING });
      }
    }
    if (csrf && !teams) {
      dispatch({ type: UPDATE_TEAMS_LOADING });
      getTeams();
    }
  }, [csrf, teams, dispatch]);

  useEffect(() => {
    const updateUserProfile = async () => {
      let res = await getCurrentUser(csrf);
      let profile =
        res.payload && res.payload.data && !res.error
          ? res.payload.data
          : undefined;

      if (profile) {
        dispatch({ type: MY_PROFILE_UPDATE, payload: profile });
      }
    };
    if (csrf && !userProfile) {
      updateUserProfile();
    }
  }, [csrf, userProfile]);

  useEffect(() => {
    const getAllMemberSkills = async () => {
      const res = await getMemberSkills(csrf);
      const memberSkills =
        res && res.payload && res.payload.data ? res.payload.data : null;
      if (memberSkills) {
        dispatch({ type: UPDATE_MEMBER_SKILLS, payload: memberSkills });
      }
    };
    if (csrf && !memberSkills) {
      getAllMemberSkills();
    }
  }, [csrf, memberSkills]);

  useEffect(() => {
    async function getMemberProfiles() {
      let res = await getAllMembers(csrf);
      let profiles =
        res.payload && res.payload.data && !res.error
          ? res.payload.data
          : undefined;

      if (profiles) {
        dispatch({ type: UPDATE_MEMBER_PROFILES, payload: profiles });
        dispatch({ type: UPDATE_PEOPLE_LOADING, payload: false });
      }
    }
    async function getTerminatedMembers() {
      let res = await getAllTerminatedMembers(csrf);
      let profiles =
        res.payload && res.payload.data && !res.error
          ? res.payload.data
          : undefined;

      if (profiles) {
        dispatch({ type: UPDATE_TERMINATED_MEMBERS, payload: profiles });
      }
    }
    if (csrf && userProfile && !memberProfiles) {
      dispatch({ type: UPDATE_PEOPLE_LOADING, payload: true });
      getMemberProfiles();
      if (selectCanViewTerminatedMembers(state)) {
        getTerminatedMembers();
      }
    }
  }, [csrf, userProfile, memberProfiles]);

  useEffect(() => {
    function getAllTheCheckins() {
      if (
        userProfile &&
        userProfile.permissions?.some(p =>
          p?.permission?.includes('CAN_VIEW_CHECKINS_REPORT')
        ) &&
        id &&
        csrf
      ) {
        getAllCheckinsForAdmin(dispatch, csrf);
      } else if (id && csrf && selectCanViewCheckinsPermission(state)) {
        getCheckins(id, pdlId, dispatch, csrf);
      }
    }
    if (csrf && !checkins) {
      getAllTheCheckins();
    }
  }, [csrf, pdlId, id, userProfile, checkins]);

  useEffect(() => {
    const getAllSkills = async () => {
      const res = await getSkills(csrf);
      const data =
        res &&
        res.payload &&
        res.payload.data &&
        res.payload.status === 200 &&
        !res.error
          ? res.payload.data
          : null;
      if (data && data.length > 0) {
        dispatch({ type: UPDATE_SKILLS, payload: data });
      }
    };
    if (csrf && !skills) {
      getAllSkills();
    }
  }, [csrf, skills]);

  useEffect(() => {
    const getAllCertifications = async () => {
      const res = await getCertifications(csrf);
      const data =
          res &&
          res.payload &&
          res.payload.data &&
          res.payload.status === 200 &&
          !res.error
              ? res.payload.data
              : null;
      if (data && data.length > 0) {
        dispatch({ type: UPDATE_CERTIFICATIONS, payload: data });
      }
    };
    if (csrf && !certifications) {
      getAllCertifications();
    }
  }, [csrf, certifications]);

  useEffect(() => {
    const getRoles = async () => {
      const res = await getAllRoles(csrf);
      const data =
        res &&
        res.payload &&
        res.payload.data &&
        res.payload.status === 200 &&
        !res.error
          ? res.payload.data
          : null;
      if (data && Array.isArray(data) && data.length > 0) {
        dispatch({ type: SET_ROLES, payload: data });
      }
    };
    if (csrf && !roles) {
      getRoles();
    }
  }, [csrf, roles]);

  useEffect(() => {
    const getUserRoles = async () => {
      // make call to the API
      let res = await getAllUserRoles(csrf);
      return res.payload &&
        res.payload.data &&
        res.payload.status === 200 &&
        !res.error
        ? res.payload.data
        : null;
    };

    if (csrf && !userRoles) {
      getUserRoles().then(userRoles => {
        dispatch({ type: SET_USER_ROLES, payload: userRoles });
      });
    }
  }, [csrf, userRoles]);

  const value = useMemo(() => {
    return { state, dispatch };
  }, [state]);
  return (
    <AppContext.Provider value={props.value || value}>
      {props.children}
    </AppContext.Provider>
  );
};

export { AppContext, AppContextProvider };
