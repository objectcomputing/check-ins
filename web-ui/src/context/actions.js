export const MY_PROFILE_UPDATE = "@@check-ins/update_profile";
export const UPDATE_USER_BIO = "@@check-ins/update_bio";
export const UPDATE_CHECKINS = "@@check-ins/update_checkins";
export const UPDATE_CHECKIN = "@@check-ins/update_checkin";
export const ADD_CHECKIN = "@@check-ins/add_checkin";
export const UPDATE_TOAST = "@@check-ins/update_toast";
export const UPDATE_MEMBER_SKILLS = "@@check-ins/update_member_skills";
export const DELETE_MEMBER_SKILL = "@@check-ins/delete_member_skill";
export const ADD_MEMBER_SKILL = "@@check-ins/add_member_skill";
export const UPDATE_TEAMS = "@@check-ins/update_teams";
export const UPDATE_MEMBER_PROFILES = "@@check-ins/update_member_profiles";
export const UPDATE_TEAM_MEMBERS = "@@check-ins/update_team_members";
export const ADD_SKILL = "@@check-ins/add-skill";
export const DELETE_SKILL = "@@check-ins/delete-skill";
export const UPDATE_SKILL = "@@check-ins/update_skill";
export const UPDATE_SKILLS = "@@check-ins/update_skills";
export const ADD_TEAM = "@@check-ins/add_team";
export const SET_CSRF = "@@check-ins/update_csrf";
export const SET_ROLES = "@@check-ins/set_roles";
