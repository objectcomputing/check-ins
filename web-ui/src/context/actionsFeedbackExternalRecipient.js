export const SET_CSRF = '@@check-ins/update_csrf';
export const UPDATE_TOAST = '@@check-ins/update_toast';

