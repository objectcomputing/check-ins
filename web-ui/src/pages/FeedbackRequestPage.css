.feedback-request-page {
    margin: 4rem 2rem 2rem 2rem;
}

.header-container {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 20px;
}