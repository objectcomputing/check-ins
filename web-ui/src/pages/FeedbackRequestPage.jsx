import React, {useState, useEffect} from "react";
import { makeStyles } from "@material-ui/core/styles";
import Stepper from "@material-ui/core/Stepper";
import Step from "@material-ui/core/Step";
import StepLabel from "@material-ui/core/StepLabel";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import { Link, useLocation, Redirect } from 'react-router-dom';
import queryString from 'query-string';
import TemplateCard from "../components/template-card/TemplateCard"
import FeedbackRecipientSelector from "../components/feedback_recipient_selector/FeedbackRecipientSelector";

import "./FeedbackRequestPage.css";
import TemplatePreviewModal from "../components/template-preview-modal/TemplatePreviewModal";

const useStyles = makeStyles((theme) => ({
  root: {
    backgroundColor: "transparent"
  },
  appBar: {
    position: "relative",
  },
  media: {
    height: 0,
  },
  expand: {
    justifyContent: "right",
    transition: theme.transitions.create("transform", {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    justifyContent: "right",
  }
}));

function getSteps() {
  return ["Select template", "Select recipients", "Set due date"];
}

function isEmpty() {
  if(window.location.href.indexOf("?from=") > -1 && (window.location.href.indexOf("step=") > -1)) {
    return false;
  }
  else {
    return true;
  }
}

function getTemplates() {
  return [
    {
      title: "Ad Hoc",
      isAdHoc: true,
      description: "Ask a single question.",
      creator: "Admin",
      questions: []
    },
    {
      title: "Survey 1",
      isAdHoc: false,
      description: "Make a survey with a few questions",
      creator: "Admin",
      questions: []
    },
    {
      title: "Feedback Survey 2",
      isAdHoc: false,
      description: "Another type of survey",
      creator: "Jane Doe",
      questions: [],
    },
    {
      title: "Custom Template",
      isAdHoc: false,
      description: "A very very very very very very very very very very very very very very very very very very very very very very very very very very long description",
      creator: "Bob Smith",
      questions: []
    },
  ];
}

const FeedbackRequestPage = () => {
  const steps = getSteps();
  const classes = useStyles();
  const location = useLocation();
  const query = queryString.parse(location?.search).step?.toString();
  let activeStep = location?.search ? parseInt(query) : 1;
  const numbersOnly = /^\d+$/.test(query);
  const [preview, setPreview] = useState({open: false, selectedTemplate: null});

  const handlePreviewOpen = (event, selectedTemplate) => {
    setPreview({open: true, selectedTemplate: selectedTemplate});
  }

  const handlePreviewClose = (selectedTemplate) => {
    setPreview({open: false, selectedTemplate: selectedTemplate});
  }

  const onCardClick = (template) => {
    if (template.isAdHoc) {
      setPreview({open: true, selectedTemplate: template});
    } else {
      console.log(`Selected ${template.title}`);
    }
  }

  if (activeStep < 1 || activeStep > steps.length || !numbersOnly) {
    return (
      <Redirect to="/feedback/request?step=1"/>
    );
  }

  return (
    <div className="feedback-request-page">
      {preview.selectedTemplate &&
        <TemplatePreviewModal
          template={preview.selectedTemplate}
          open={preview.open}
          onClose={handlePreviewClose}
        />
      }

      <div className="header-container">
        <Typography variant="h4">Feedback Request for <b>John Doe</b></Typography>
        <div>
            <div>
              <Link
                className={`no-underline-link ${activeStep <= 1 ? 'disabled-link' : ''}`}
                to={`?step=${activeStep - 1}`}
              >
                <Button
                  disabled={activeStep <= 1}>
                  Back
                </Button>
              </Link>

              <Link
                className={`no-underline-link ${activeStep > getSteps().length ? 'disabled-link no-underline-link' : ''}`}
                to={activeStep===3 ?`/feedback/request/confirmation` : `?step=${activeStep + 1}`}>
                <Button
                  disabled={activeStep > getSteps().length || (activeStep === 2 && isEmpty())}
                  variant="contained"
                  color="primary">
                  {activeStep === steps.length ? "Submit" : "Next"}
                </Button>
              </Link>
            </div>
        </div>
      </div>
      <Stepper activeStep={activeStep - 1} className={classes.root}>
        {steps.map((label) => {
          const stepProps = {};
          const labelProps = {};
          return (
            <Step key={label} {...stepProps}>
              <StepLabel {...labelProps} key={label}>{label}</StepLabel>
            </Step>
          );
        })}
      </Stepper>
      <div className="current-step-content">
          {activeStep === 1 && <TemplatePreviewModal/> &&
          <div className="card-container">
            {getTemplates().map((template) => (
              <TemplateCard
                title={template.title}
                creator={template.creator}
                description={template.description}
                isAdHoc={template.isAdHoc}
                questions={template.questions}
                expanded={preview.open}
                onClick={(e) => handlePreviewOpen(e, template)}
                onCardClick={() => onCardClick(template)}/>
            ))}
          </div>
        }
        {activeStep === 2 && <FeedbackRecipientSelector />}
      </div>
    </div>
  );
}

export default FeedbackRequestPage;