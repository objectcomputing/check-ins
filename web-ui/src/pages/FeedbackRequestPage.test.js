import React from "react";
import FeedbackRequestPage from "./FeedbackRequestPage";
import {AppContextProvider} from "../context/AppContext";
import {BrowserRouter} from "react-router-dom";

it("renders correctly", () => {
  snapshot(
      <AppContextProvider>Ëš
        <BrowserRouter>
          <FeedbackRequestPage/>
        </BrowserRouter>
      </AppContextProvider>
  );
});