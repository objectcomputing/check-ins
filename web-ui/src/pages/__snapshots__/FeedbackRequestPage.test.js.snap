// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  "Ëš",
  <div
    className="feedback-request-page"
  >
    <div
      className="header-container"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4"
      >
        Feedback Request for 
        <b>
          John Doe
        </b>
      </h4>
      <div>
        <div>
          <a
            className="no-underline-link disabled-link"
            href="/feedback/request?step=0"
            onClick={[Function]}
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text Mui-disabled Mui-disabled"
              disabled={true}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Back
              </span>
            </button>
          </a>
          <a
            className="no-underline-link "
            href="/feedback/request?step=2"
            onClick={[Function]}
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Next
              </span>
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiStepper-root MuiStepper-horizontal makeStyles-root-1 MuiPaper-elevation0"
    >
      <div
        className="MuiStep-root MuiStep-horizontal"
      >
        <span
          className="MuiStepLabel-root MuiStepLabel-horizontal"
        >
          <span
            className="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                className="MuiStepIcon-text"
                textAnchor="middle"
                x="12"
                y="16"
              >
                1
              </text>
            </svg>
          </span>
          <span
            className="MuiStepLabel-labelContainer"
          >
            <span
              className="MuiTypography-root MuiStepLabel-label MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
            >
              Select template
            </span>
          </span>
        </span>
      </div>
      <div
        className="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          className="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        className="MuiStep-root MuiStep-horizontal"
      >
        <span
          className="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
        >
          <span
            className="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiStepIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                className="MuiStepIcon-text"
                textAnchor="middle"
                x="12"
                y="16"
              >
                2
              </text>
            </svg>
          </span>
          <span
            className="MuiStepLabel-labelContainer"
          >
            <span
              className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              Select recipients
            </span>
          </span>
        </span>
      </div>
      <div
        className="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          className="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        className="MuiStep-root MuiStep-horizontal"
      >
        <span
          className="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
        >
          <span
            className="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiStepIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                className="MuiStepIcon-text"
                textAnchor="middle"
                x="12"
                y="16"
              >
                3
              </text>
            </svg>
          </span>
          <span
            className="MuiStepLabel-labelContainer"
          >
            <span
              className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              Set due date
            </span>
          </span>
        </span>
      </div>
      <div
        className="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          className="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        className="MuiStep-root MuiStep-horizontal"
      >
        <span
          className="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
        >
          <span
            className="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiStepIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                className="MuiStepIcon-text"
                textAnchor="middle"
                x="12"
                y="16"
              >
                4
              </text>
            </svg>
          </span>
          <span
            className="MuiStepLabel-labelContainer"
          >
            <span
              className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              Done!
            </span>
          </span>
        </span>
      </div>
    </div>
    <div
      className="current-step-content"
    >
      <div
        className="card-container"
      >
        <div
          className="MuiPaper-root MuiCard-root feedback-template-card MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            className="MuiCardContent-root"
          >
            <div
              className="card-content"
            >
              <div>
                <div
                  className="template-name"
                >
                  Ad Hoc
                </div>
                <div
                  className="description-and-creator"
                >
                  <div
                    className="description"
                  >
                    Send a single question
                  </div>
                </div>
              </div>
              <div
                className="creator-wrapper"
              >
                Created by:
                <div
                  className="creator"
                >
                  Admin
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiPaper-root MuiCard-root feedback-template-card MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            className="MuiCardContent-root"
          >
            <div
              className="card-content"
            >
              <div>
                <div
                  className="template-name"
                >
                  Survey 1
                </div>
                <div
                  className="description-and-creator"
                >
                  <div
                    className="description"
                  >
                    Make a survey of a few questions
                  </div>
                </div>
              </div>
              <div
                className="creator-wrapper"
              >
                Created by:
                <div
                  className="creator"
                >
                  Admin
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiPaper-root MuiCard-root feedback-template-card MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            className="MuiCardContent-root"
          >
            <div
              className="card-content"
            >
              <div>
                <div
                  className="template-name"
                >
                  Feedback Survey 2
                </div>
                <div
                  className="description-and-creator"
                >
                  <div
                    className="description"
                  >
                    Another type of survey
                  </div>
                </div>
              </div>
              <div
                className="creator-wrapper"
              >
                Created by:
                <div
                  className="creator"
                >
                  Jane Doe
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiPaper-root MuiCard-root feedback-template-card MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            className="MuiCardContent-root"
          >
            <div
              className="card-content"
            >
              <div>
                <div
                  className="template-name"
                >
                  Custom Template
                </div>
                <div
                  className="description-and-creator"
                >
                  <div
                    className="description"
                  >
                    A very very very very very very very very very very very very very very very very very ver...
                  </div>
                </div>
              </div>
              <div
                className="creator-wrapper"
              >
                Created by:
                <div
                  className="creator"
                >
                  Bob Smith
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
